From 5100e27117e33ef047da59da7e59da609f0835bb Mon Sep 17 00:00:00 2001
From: Patrick Lauer <patrick@gentoo.org>
Date: Wed, 30 Jul 2025 02:17:27 +0300
Subject: [PATCH 06/26] [ffmpeg] fix building with libav-10

Bug: https://bugs.gentoo.org/538500
Origin: https://github.com/gentoo/gentoo-historical-2/commit/b1a41c15d12c318baf4c2964575b20aa41399f61
Original-commuter: Mike Frysinger <vapier@gentoo.org>
Patch-name: transcode-1.1.7-libav-10.patch
See-also: https://wiki.libav.org/Migration/10
Signed-off-by: Alexander Golubev <fatzer2@gmail.com>
---
 filter/filter_resample.c | 3 ++-
 import/probe_ffmpeg.c    | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/filter/filter_resample.c b/filter/filter_resample.c
index f94edf1b..c94c6cce 100644
--- a/filter/filter_resample.c
+++ b/filter/filter_resample.c
@@ -37,6 +37,7 @@
 #include "libtc/optstr.h"
 #include "libtc/tcavcodec.h"
 #include "libtc/tcmodule-plugin.h"
+#include <libavresample/avresample.h>
 
 
 typedef struct {
@@ -45,7 +46,7 @@ typedef struct {
 
     int bytes_per_sample;
 
-    ReSampleContext *resample_ctx;
+    AVAudioResampleContext *resample_ctx;
 } ResamplePrivateData;
 
 static const char resample_help[] = ""
diff --git a/import/probe_ffmpeg.c b/import/probe_ffmpeg.c
index bb2ad24a..3fc2e34a 100644
--- a/import/probe_ffmpeg.c
+++ b/import/probe_ffmpeg.c
@@ -51,8 +51,8 @@ static void translate_info(const AVFormatContext *ctx, ProbeInfo *info)
             info->bitrate = st->codec->bit_rate / 1000;
             info->width = st->codec->width;
             info->height = st->codec->height;
-            if (st->r_frame_rate.num > 0 && st->r_frame_rate.den > 0) {
-                info->fps = av_q2d(st->r_frame_rate);
+            if (st->avg_frame_rate.num > 0 && st->avg_frame_rate.den > 0) {
+                info->fps = av_q2d(st->avg_frame_rate);
             } else {
                 /* watch out here */
                 info->fps = 1.0/av_q2d(st->codec->time_base);
-- 
2.49.1

